test/rx_ext/repeater_test.dart

```diff
@@ -0,0 +1,134 @@
+import 'dart:async';
+
+import 'package:flutter_reactive_ble/src/rx_ext/repeater.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+void main() {
+  group("Repeater", () {
+    StreamController<int> underlyingStreamController;
+    _RepeaterHandlerMock<int> handler;
+    Repeater<int> sut;
+
+    setUp(() {
+      underlyingStreamController = StreamController(sync: true);
+      handler = _RepeaterHandlerMock();
+
+      when(handler.onListenEmitFrom())
+          .thenAnswer((_) => underlyingStreamController.stream);
+    });
```

Comment:
Ouch. The SUT is declared at _this_ level, but initialized. That make the variable useless at _this_ level (if not dangerous, because it is effectively uninitialized [yes, it's "initialized" to `null` but do you expect it to be null in SUT?]).